ambry-api/src/main/java/com/github/ambry/named/NamedBlobDb.java

```diff
@@ -24,6 +23,39 @@
  */
 public interface NamedBlobDb {
 
+  /**
+   * The mode for get method.
+   */
+  enum GetMode {
+    /**
+     * Return NamedBlobRecord in get method when it's not expired or deleted.
+     */
+    Include_None,
+    /**
+     * Return NamedBlobRecord in get method even if it's deleted.
+     */
+    Include_Deleted,
+    /**
+     * Return NamedBlobRecord in get method even if it's expired.
+     */
+    Include_Expired,
+    /**
+     * Return NamedBlobRecord in get method even if it's deleted or expired.
+     */
+    Include_All
+  }
+
+  /**
+   * Look up a {@link NamedBlobRecord} by name.
+   * @param accountName the name of the account.
+   * @param containerName the name of the container.
+   * @param blobName the name of the blob.
+   * @param mode The {@link GetMode} for this get method.
+   * @return a {@link CompletableFuture} that will eventually contain either the {@link NamedBlobRecord} for the named
+   *         blob or an exception if an error occurred.
+   */
+  CompletableFuture<NamedBlobRecord> get(String accountName, String containerName, String blobName, GetMode mode);
```

Comment:
Can we directly use getOption instead of the new generated GetMode enum. It sort of duplicate the code, and in the future if the getOption changed, we need to remember to change this enum accordingly avoid confusion, which is sort of an anti pattern to me.

Reply:
The reason why I want to have a different Get option in named blob is because I don't want the named blob db relying on http request related object, like different component has different set of errors.

But I suppose if it's one to one mapping and GetOption is generic enough, we can re-use it here.

Will update.